spec/auth_helper.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+RSpec.configure do |config|
+  config.before(:each) do
+    login = { user: 'user', password: 'password' }
+    allow(Rails.application.credentials).to receive(:login).and_return(login)
```

Comment:
Maybe there is a cleaner way to do that without method stubs. We could e.g. put public secrets for testing into `secrets.yml`.

Reply:
In case you need to change the tested code to make that happen, leave it as it is.

Reply:
Yes, tried that as well, but it doesn’t work as secrets and credentials cannot be used interchangeably. We could have [separate credentials for different environments](https://github.com/rails/rails/pull/33521), though.